httpie_aws_authv4.py

```diff
@@ -99,39 +114,66 @@ class AWSv4AuthPlugin(AuthPlugin):
     auth_parse = False
     prompt_password = False
 
+    alias_params = {
+        "ak": "access_key",
+        "sk": "secret_key",
+        "p": "profile",
+        "d": "domain",
+        "s": "service",
+        "r": "region",
+    }
+    """map of alias -> full param name"""
+
     def get_auth(self, username=None, password=None):
-        # To remain consistant with AWS tools, boto3 credential store is used
-        #   to retrieve the users API keys.  This means environment variables
-        #   take precedent over ~/.aws/credentials and IAM roles.  credentials
-        #   can be provided on the CLI using the -a flag
-        #       eg, -a <access_key>:<secret_access_key>
+        """
+        To remain consistant with AWS tools, boto3 credential store is used
+        to retrieve the users API keys.  This means environment variables
+        take precedent over ~/.aws/credentials and IAM roles.  credentials
+        can be provided on the CLI using the -a flag eg,
+            -a <access_key>:<secret_access_key>
+
+        You can also provide aditional parameters using following syntax:
+            -a access_key=<access_key>,secret_key=<secret_key>,
+            profile=<profile>,domain=<domain>,service=<service>,region=<region>
 
-        # An attempt is made to try and determine the region, endpoint and
-        #   service from a given request URL.  If you are using a custom domain
-        #   this is no doubt going fail and the user will need to input the neccessary
-        #   information using the `-a` flag.
+        or short equvivalent:
+            -a ak=<access_key>,sk=<secret_key>,p=<profile>,d=<domain>,
+            s=<service>,r=<region>
 
-        # The format for the `-a` flag is:
-        #   "service:region:endpoint"
+        For example:
+            -a s=executeapi,r=eu-west-1
 
-        access_key = None
-        secret_key = None
-        domain = None
-        profile = None
+        An attempt is made to try and determine the region, endpoint and
+        service from a given request URL if you are not supplying this data as above.
+        If you are using a custom domain this will fail and service plus region need
+        to be supplied in the -a flag.
+        """
+
+        params: dict
 
         if self.raw_auth is not None:
-            parts = self.raw_auth.split(":")
-            if len(parts) >= 2:
-                if parts[0] == "profile":
-                    profile = parts[1]
+
+            try:
+                if '=' in self.raw_auth:
+
+                    params = dict(
+                        x.split('=', 2) for x in cast('str', self.raw_auth).split(',')
```

Comment:
Out of curiosity, why was `cast` required here?

Reply:
Oh, it's not required, I was just looking into a thing with the split and forgot to remove the cast.